src/App/Pages/Settings/SettingsPage/SettingsPageViewModel.cs

```diff
@@ -621,5 +635,14 @@ private VaultTimeoutAction GetVaultTimeoutActionFromKey(string key)
         private string CreateSelectableOption(string option, bool selected) => selected ? $"✓ {option}" : option;
 
         private bool CompareSelection(string selection, string compareTo) => selection == compareTo || selection == $"✓ {compareTo}";
+
+        public async Task SetScreenCaptureAllowedAsync()
+        {
+            var _screenCaptureCurrentState = _screenCaptureListItem.SubLabel.Equals(AppResources.Enabled);
+            await _stateService.SetScreenCaptureAllowedAsync(!_screenCaptureCurrentState);
+            _screenCaptureAllowed = !_screenCaptureCurrentState;
+            await _deviceActionService.SetScreenCaptureAllowedAsync();
+            BuildList();
```

Comment:
@fedemkr My approach here was to avoid toggling with _screenCaptureAllowed, because  multiple threads may try to execute this code at the same time for some reason, and cause an inconsistent state.